guides/source/association_basics.md

```diff
@@ -537,6 +537,70 @@ end
 
 ![Polymorphic Association Diagram](images/association_basics/polymorphic.png)
 
+### Associations between Models with Composite Primary Keys
+
+Rails is often able to infer the primary key - foreign key information between associated models with composite
+primary keys without needing extra information. Take the following example:
+
+```ruby
+class Order < ApplicationRecord
+  self.primary_key = [:shop_id, :id]
+  has_many :books
+end
+
+class Book < ApplicationRecord
+  belongs_to :order
+end
+```
+
+Here, Rails assumes that the `:id` column should be used as the primary key for the association between an order
+and its books, just as with a regular `has_many` / `belongs_to` association. It will infer that the foreign key column
+on the `books` table is `:order_id`. Accessing a book's order:
+
+```ruby
+order = Order.create!(id: [1, 2], status: "pending")
+book = order.books.create!(title: "A Cool Book")
+
+book.reload.order
+```
+
+will generate the following SQL to access the order:
+
+```sql
+SELECT "orders".* FROM "orders" WHERE "orders"."id" = ? LIMIT ?  [["id", 2], ["LIMIT", 1]]
```

Comment:
I believe we're trying to use conceptual cleaned up SQL for these. So more like:

```sql
SELECT * FROM order WHERE id = 2
```

And at least in the active_record_basics guide we say its "equivalent" to the SQL rather than generating it, probably because we're taking those liberties.

Reply:
Yes, let's change this.

Reply:
Oh, thank you, didn't realize that!